pandapipes/io/convert_format.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2016-2020 by University of Kassel and Fraunhofer Institute for Energy Economics
+# and Energy System Technology (IEE), Kassel. All rights reserved.
+
+from packaging import version
+
+from pandapipes import __version__
+
+
+def convert_format(net):
+    """
+    Converts old nets to new format to ensure consistency. The converted net is returned.
+    """
+    if isinstance(net.version, str) and version.parse(net.version) >= version.parse(__version__):
+        return net
+    _rename_columns(net)
+    _update_initial_run(net)
+    reset_results(net)
+    net.version = __version__
+    return net
+
+
+def _rename_columns(net):
+    if "controller" in net:
+        if ("controller" in net.controller) and ("object" in net.controller):
+            if net['controller'].at[0, 'object'] is None:
+                net['controller'].drop('object', inplace=True, axis=1)
+            else:
+                net['controller'].drop('controller', inplace=True, axis=1)
+        net["controller"].rename(columns={"controller": "object"}, inplace=True)
+
+
+def _update_initial_run(net):
+    if "controller" in net:
+        for ctrl in net.controller.object.values:
+            if hasattr(ctrl, 'initial_pipeflow'):
+                ctrl.initial_run = ctrl.initial_pipeflow
+
+
+def reset_results(net):
```

Comment:
Does that mean that results will always be removed in old versions? I think that is not what was intended in pandapower. We want to keep results inside the tables as much as possible.

Reply:
Yes, right, pandapower is doing the same in my opinion. When I load for example:
net = nw.mv_oberrhein() and run reset_results from pandapower, I delete all results.
I think that it should make sure that the res_tables are always comprise the correct columns. As in our cases the res_tables are created differently, we could think about keeping the results.

Reply:
Actually my comment refers rather to the convert_format function. There you state that if the version in net is smaller than the current pandapipes version, the convert-functions will be called. In that case, reset_results will be called for all older versions, although there is no conversion problem.

Reply:
I see what you mean. I would remove it then!